Adding Seperate Line Colour and Owner Colour for Player A and Player B

AliHassan245 opened the pull request:
Fix #29

AliHassan245 commented:
@quozl please have a look at my PR which implements seperate line colour and owner colour for player A and player B.

quozl commented:
https://github.com/sugarlabs/fractionbounce/pull/20 comment applies here too.

We only have dual play in this activity because it lacks collaboration features of other Sugar activities.  I don't like the idea of adding colours for players given that our collaboration feature uses profile colour of the player user.

AliHassan245 commented:
Yes that's true. The way I am thinking is that apart from default line color set to profile colour. We already have colors choice in toolbar for both players that's why we should give option to both players to modify their color and this customized color persists on turn rotation.

quozl commented:
Please rebase and squash the commits, and write a proper commit message, thanks.

Commit:
Implement Player-Specific Line and Owner Colors on Turn Switching

- Initialize separate line colors for Player A and Player B (defaulting to black).
- Update set_line_color() to modify the line color for only the current player.
- Modify draw_line() to display lines using the respective player’s chosen color.
- Adjust draw_current_player() to add logic to dynamically switch the drawing color based on the current player's turn.
- Update box.showOwner() to display the showOwner text in the corresponding player's line color.
